crates/sqlbuiltins/src/functions/table/arr_json/exec.rs

```diff
@@ -0,0 +1,262 @@
+                let mut arr = UInt64Builder::new();
+                for row in rows.iter() {
+                    let val: Option<u64> = row[field.name()].as_u64();
+                    arr.append_option(val);
+                }
+                Arc::new(arr.finish())
+            }
+            DataType::Boolean => {
+                let mut arr = BooleanBuilder::new();
+                for row in rows.iter() {
+                    let val: Option<bool> = row[field.name()].as_bool();
+                    arr.append_option(val);
+                }
+                Arc::new(arr.finish())
+            }
+            DataType::Utf8 => {
+                // Assumes an average of 16 bytes per item.
+                let mut arr = StringBuilder::with_capacity(rows.len(), rows.len() * 16);
+                for row in rows.iter() {
+                    let val: Option<&str> = row[field.name()].as_str();
+                    arr.append_option(val);
+                }
+                Arc::new(arr.finish())
+            }
+            // TODO
+            // 1. Nested Lists and Structs
+            // 2. add timestamp and date conversions; (serde does not suppoert that though)
+            other => {
+                return Err(ArrowError::CastError(format!(
+                    "Failed to convert {:#?}",
+                    other
+                )))
+            }
+        };
+        columns.push(col);
+    }
+    let batch = RecordBatch::try_new(schema, columns)?;
+    Ok(batch)
+}
+
+/// A [`FileOpener`] that opens an Array JSON file and yields a [`FileOpenFuture`]
+struct ArrayJsonOpener {
+    batch_size: usize,
+    projected_schema: SchemaRef,
+    file_compression_type: FileCompressionType,
+    object_store: Arc<dyn ObjectStore>,
+}
+
+impl ArrayJsonOpener {
+    /// Returns an  [`ArrayJsonOpener`]
+    pub fn new(
+        batch_size: usize,
+        projected_schema: SchemaRef,
+        file_compression_type: FileCompressionType,
+        object_store: Arc<dyn ObjectStore>,
+    ) -> Self {
+        Self {
+            batch_size,
+            projected_schema,
+            file_compression_type,
+            object_store,
+        }
+    }
+}
+
+impl FileOpener for ArrayJsonOpener {
+    fn open(&self, file_meta: FileMeta) -> Result<FileOpenFuture> {
+        let store = self.object_store.clone();
+        let schema = self.projected_schema.clone();
+        let batch_size = self.batch_size;
+        let file_compression_type = self.file_compression_type.to_owned();
+        Ok(Box::pin(async move {
+            let r = store.get(file_meta.location()).await?;
+            let stream = stream! {
+                 match r.payload {
+                GetResultPayload::File(file, _) => {
+                    let decoder = file_compression_type.convert_read(file)?;
+                    let mut reader = BufReader::new(decoder);
+                    let values: Value = serde_json::from_reader(&mut reader)
```

Comment:
I'm partial to using using simd-json here. It's generally faster out of the box for reads, and it has a [known-key](https://github.com/simd-lite/simd-json#known-key) feature that makes it even more performant when reading with a known set of keys. I also had much success using it as the base for the json readers in polars

Reply:
This is interesting. And it is compatible with `serde` out of the box. 
Thank youuu!.